### DISCUSSION TITLE
What is the correct way to quote a conditional?

### DISCUSSION THREAD

**Discussion by scottwn** [community/community#114150] [Actions] [answered]
### Select Topic Area

Question

### Body

I have a conditional in my workflow.
```
if: ! fromJson(steps.jira.outputs.response).fields.resolution && env.PROJECT == 'QNO'
```
This fails with the error, `While parsing a tag, did not find expected tag URI.` I assume this is because of the reserved `!`. So I tried to re-write it,
```
if: '! fromJson(steps.jira.outputs.response).fields.resolution && env.PROJECT == "QNO"'
```
This fails again with a different error, `Unexpected symbol: '"QNO"'` I assume this is because I can't use double quotes for string comparison. (https://docs.github.com/en/actions/learn-github-actions/expressions, "Wrapping with double quotes (") will throw an error.") So I tried it again with no quotes around `QNO`,
```
if: '! fromJson(steps.jira.outputs.response).fields.resolution && env.PROJECT == QNO'
```
This also fails with the error, `Unrecognized named-value: 'QNO'`.

What is the correct way to quote this conditional?

**Comment by ste-xx** [comment, marked as answer]
This should work: 
```
if: ${{ !fromJson(steps.jira.outputs.response).fields.resolution && env.PROJECT == 'QNO' }}
```

Example workflow:
https://github.com/ste-xx/example-gh-wf/blob/main/.github/workflows/conditional.yml
